site-builder/src/site/resource.rs

```diff
@@ -348,22 +396,42 @@ impl ResourceManager {
                 .expect("the path should not terminate in `..`"),
         );
 
-        let content_type =
-            match ContentType::try_from_extension(extension.to_str().ok_or(anyhow!(
-                "Could not convert the extension {:?} to a string.",
-                extension.to_string_lossy()
-            ))?) {
-                Ok(content_type) => content_type,
-                Err(_) => {
-                    tracing::warn!(
-                        "The extension {} string for file {} could not be decoded.
-                        Defaulting to arbitrary binary content type: octet-stream.",
-                        extension.to_string_lossy(),
-                        full_path.to_string_lossy()
-                    );
-                    ContentType::ApplicationOctetstream // arbitrary binary data RFC 2046
-                }
-            };
+        // Get the headers of the specific resource if they exist
+        // for this specific path.
+        // 1. Is ws-config.json specified?
+        let all_headers = match &self.ws_config {
+            Some(config) => &config.headers,
+            None => return Ok(None),
+        };
+        // 2. Are "headers" included in ws-config?
+        let resource_path = full_path_to_resource_path(full_path, root)?;
+        let headers = match all_headers {
+            Some(map) => map.get(&resource_path),
+            _ => None,
+        };
+        // 3. Does the specific resource has headers defined for it? If not, use defaults.
+        let http_headers = match headers {
+            Some(h) => HttpHeaders::try_from(h.clone()),
+            _ => Ok(HttpHeaders::new()),
+        };
+
```

Comment:
Important: I think that, if there are no content type headers, we should infer it from the extension